### DISCUSSION TITLE
How to embed WPF User Control ?

### DISCUSSION THREAD

**Discussion by xuyue** [AvaloniaUI/Avalonia#7665] [Q&A] [answered]
Excuse me, I need to embed WPF window or windows application under windows, how should I do it?

**Comment by timunie** [comment, marked as answer]
Hi @xuyue 

I think you need to use a `NativeControlHost` for this. 

API: http://reference.avaloniaui.net/api/Avalonia.Controls/NativeControlHost/ 
Sample: https://github.com/AvaloniaUI/Avalonia/tree/master/samples/interop/NativeEmbedSample
See also: https://github.com/AvaloniaUI/Avalonia/discussions/7550

I hope this helps. 

Happy coding
Tim

**Reply by xuyue** [reply]
thanks

**Reply by derekantrican** [reply]
FYI: looks like the sample was removed, but it can still be seen at https://github.com/AvaloniaUI/Avalonia/blob/b045393df3931e33d4fbefc8ff3ea28068386014/samples/interop/NativeEmbedSample/EmbedSample.cs

**Comment by Mrxx99** [comment]
There is also this great article that specifically covers WPF:
https://www.codeproject.com/Articles/5348155/Embedding-Native-Windows-and-Linux-Views-Controls
